crypto/evp/scrypt.c

```diff
@@ -139,7 +139,7 @@ static void scryptROMix(unsigned char *B, uint64_t r, uint64_t N,
         *pV = *pB++;
         *pV |= *pB++ << 8;
         *pV |= *pB++ << 16;
-        *pV |= *pB++ << 24;
+        *pV |= (uint32_t)*pB++ << 24;
```

Comment:
As for "two's complement". You have to read it in context. Original "it doesn't *seem* to affect" is nothing else but empirical statement, i.e. based on experimental observation. Then it's modified to "it doesn't affect the result on any of the supported platforms", i.e. no "seem", but reference to "supported platforms". This is also nothing else but empirical statement, just stronger one. Note that quotes' placement was not a coincidence, because "all of them are two's complement" is not actually integral part of the statement, but suggested explanation for how come statement is supported by experimental observations. In other words, I'm not suggesting to make unsafe assumptions, only try to explain how it worked so far.